### DISCUSSION TITLE
Ability to set Dashboard defaults and visibility per user (not per device or maybe in addition to)

### DISCUSSION THREAD

**Discussion by u8915055** [home-assistant/frontend#11278] [Other feature requests]
## The request

I am looking for a way, now with the multiple dashboard capability, to have more control over the multiple dashboard capabilities per user.  I was hoping for an ability to define a default dashboard per user and also to change visibility (or even access) to certain dashboards per user.  I would also like the ability to store this info persistently perhaps in the auth file in .storage per user so that regardless of what device, or whether the user logs out or in, they always get that dashboard.  It would also be really useful to be able to rename the 'Overview' dashboard.

I have played around with an integration called 'custom-sidebar' which gives me the ability to re-order dashboards in the sidebar, create new ones as hrefs to pages, and to hide but it doesnt give me the ability to set defaults.  So even if I hide the 'Overview' dashboard, when a new user logs in  they may not see the 'Overview' dashboard but they'll be directed to lovelace/0 anyway.

**Comment by zombiehoffa** [comment]
This is pretty much necessary for anyone with more than 2 people living in a house so I really hope somebody implements it.

**Comment by Squale76** [comment]
In my installation, I have several users with, for each, an associated dashboard.
When a user logs out, he loses this association.
It should be possible, via the admin account, to associate a default dashboard to each user without going to each device by logging in with each account

**Comment by mdehaas** [comment]
I would even suggest that this could be implemented in two stages:
1) Allow to set a dashboard for a user under Configuration > Users, this dashboard will be used as their default dashboard. You should set all these dashboards to not show in the sidebar so the default dashboard would be virtual... This would be quite straight forward and could be implemented using the current authorisation possibilities
2) Allow to configure permissions at the dashboard level which would need a more capable and advanced rights system in which there are N to M possibilities for connecting users and resources, and even a third dimension to define the access level to the dashboard. This would also need to have access levels to al sensors, switches, etc. to allow a user to change the dashboard without access to the wrong resources.

**Comment by trsqr** [comment]
I'm setting up a system for my parents. I don't want them to see the default dashboard at all. They should only see a custom dashboard I've created. 

I don't see the logic in clearing the default dashboard after a log out from the system. Isn't the point of something being default that it's the one that's shown to the user right after logging in?

So yes, I'd be very happy to see this being implemented. Additionally I'd like to see "Change the order and hide items from the sidebar" to persist between logins.

**Comment by baylanger** [comment]
Until this feature is partially or fully implemented, I’m looking for a workaround.

For that reason, I started another discussion #11316 - please chip in if you have any comments to add.

Thank you!

**Comment by jandr** [comment]
I would also like to see the option for users to define which default dashboard they want to choose (among the ones available to them).  However, I have not seen a way for non-admin users to define their default dashboards. These non-admin users do not have access to the Dashboards configuration screen (which is expected). For admins, this screen allows opening a specific dashboard and setting it as "default for this device". However, non-admin users cannot chose their default Lovelace card in this way, as they don't have access to that configuration.

I have also not seen a way for an admin user to define which Dashboard should be the default for other non-admin users' devices (or profiles, as this discussion seeks).

The only way I have seen to go around this is for an admin to set the non-admin user temporarily to "Admin", then on that user's device, access the Dashboards configuration panel, open the desired dashboard, and choose "Select as Default for this device". Then, from the admin's account, revert that user back to non-admin. 

If my suggestion, being slightly different than what this discussion is mainly aimed at, should be brought up in a separate discussion thread, please let me know and I will move it there!

**Reply by sdrapha** [reply]
For the first part of your problem, that is not actually true, a non-admin user have access to set its default dashboard(on that device) by going into his profile setting, on the lower left corner of the sidebar.
You don't need to do the whole admin thing.

**Comment by sdrapha** [comment]
I agree and I'm eager to be able to set default for Me and other users, regardless of device, it should be tied to the user, not device. And if you have a wallpanel that should have a different default, just create a new user for each wallpanel ( that is a good thing for security ).

Also, even more important than setting the default, the ability to organize the order of dashboards from the backend.
When I started in HomeAssistant, I had many views inside a single dashboard, that dashboard raw file reached 17k lines. So I splitted all the views in many dashboards, now I have 32 dashboards, and it's a pain to deal with the order of them, each device has it's own. I want to organize them from the backend and have consistency.

**Comment by kongo09** [comment]
Feels like being connected to #7361 and #11778 

Goal would be a streamlined dashboard concept including authentication, user defaults and converting existing content to cards.

**Comment by Herculezz55** [comment]
Has this been implemented?

**Comment by sdrapha** [comment]
Nope. Still an issue. 
I rely on custom browser_mod as a partial workaround.

**Comment by zombiehoffa** [comment]
I am pretty surprised this is still an issue. It is the obvious glaring hole in necessary functionality and every release where this isn't refactored to support this just incurs further technical debt that will need to be overcome when they finally get around to fixing this.

**Comment by Fenrir88** [comment]
I still need this implemented.  Is there a feature roadmap to track this? Given how users can still bypass visibility control to tabs this is even more important.

**Comment by zgehr** [comment]
Running into the same issue as a new HA user - wasted to so much time trying to figure this out.  At least I know it's a browser thing now, so back to logging in my daughter and wife's phones to see if it is working right, I guess.

**Comment by tmjansson** [comment]
On user config it would be nice to be able to set default dashboard and which dashboards are visible on the side bar.
If not set the default behavoiur should apply.

**Comment by BoneheadFraggle** [comment]
You can hide most things with https://github.com/NemesisRE/kiosk-mode. That is user-based (among a lot of other possibilities).
You can hide the header, sidebar, menu button, search and a lot more completely, 

That, combined with setting visibility per user on different views in the same dashboard takes you almost all the way. 
But it is security by obscurity, as soon as the daughter discovers typing 'e' on the keyboard, she can find stuff she's not supposed to.

But for a "normal mobile user" I'd say it works pretty well. My family hasn't found the hidden stuff behind the curtains yet, and they are daily users of HA for 3 years.

**Comment by ocervinka** [comment]
FYI: PR [#25702](https://github.com/home-assistant/frontend/pull/25702) looks like it addresses the per-user default dashboard feature discussed here. Still in draft, but seems to be moving in the right direction. 👍
